assets/js/base/context/hooks/use-store-products.ts

```diff
@@ -34,8 +44,8 @@ export const useStoreProducts = ( query ) => {
 		query,
 	} );
 	return {
-		products,
-		totalProducts: parseInt( totalProducts, 10 ),
+		products: products as ProductResponseItem[], // TODO: Remove this once getCollection selector and resolver is converted to TS.
```

Comment:
On line [38 above](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5668/files#diff-badb52abe6c3543682e0a447fe296a3bbbee37d80b05d5e6e79685963e76b860R38) we call the `useCollection` hook, which looks up some data from a store and returns it. We need to convert quite a few other things to TS in order to get the correct types here (`useCollection` hook, some selectors and resolvers in the data folder, and it's not an easy task as most of these return generic nested objects so it needs a little more thought. I didn't have the time to convert everything needed by the  `useStoreProducts` hook in order to return the correct types from here, so I cast it to an array of `ProductResponseItem` which we are expecting here until we get to convert all other relevant hooks and state related code to TS.